src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileMetadataService.scala

```diff
@@ -111,22 +151,31 @@ class FileMetadataService(fileMetadataRepository: FileMetadataRepository,
         propertyName -> FileMetadataUpdate(metadataIdsToUpdate.toSeq, propertyName, propertyValue, Timestamp.from(timeSource.now), userId)
     }
 
-    PropertyUpdates(propertiesRowsToAdd, propertiesRowsToUpdate)
+    PropertyUpdates(metadataIdsToDelete, propertiesRowsToAdd, propertiesRowsToUpdate)
   }
 
-  private def updateFileMetadata(propertyUpdates: PropertyUpdates): Future[Unit] = {
+  private def updateFileMetadata(consignmentId: UUID, propertyUpdates: PropertyUpdates): Future[Unit] = {
+    val metadataToDelete: FileMetadataDelete = propertyUpdates.metadataToDelete
     val propertiesRowsToAdd: Seq[FilemetadataRow] = propertyUpdates.rowsToAdd
     val propertiesRowsToUpdate: Map[String, FileMetadataUpdate] = propertyUpdates.rowsToUpdate
+    val deleteFileMetadata: Future[Int] = fileMetadataRepository.deleteFileMetadata(metadataToDelete.fileIds, metadataToDelete.propertyNamesToDelete)
     val addFileMetadata: Future[Seq[FilemetadataRow]] = fileMetadataRepository.addFileMetadata(propertiesRowsToAdd)
     val updateFileMetadataProperties: Future[Seq[Int]] = fileMetadataRepository.updateFileMetadataProperties(propertiesRowsToUpdate)
 
     for {
+      deletedRows <- deleteFileMetadata
       _ <- addFileMetadata
       updatedRows <- updateFileMetadataProperties
     } yield {
+      val totalRowsDeleted: Int = deletedRows
```

Comment:
Do you need this variable? Can you just rename `deletedRows` to `totalRowsDeleted` ?